Core/Src/main.c

```diff
@@ -95,10 +103,24 @@ int main(void)
   /* USER CODE BEGIN WHILE */
   while (1)
   {
+	  HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_RESET); //setting the PB4 port (I think that's the GPIO port?) low
+
+	  HAL_SPI_TransmitReceive(&hspi1, T_buffer, R_buffer, 8, 500);
+	  adc_value = (R_buffer[1] << 8) | R_buffer[2];
```

Comment:
For `adc_value` this is correct but since you only want the first 10 bits, you should "mask" the adc value by 0b00000011 11111111 OR 0x3FF by doing:

`adc_value = ((R_buffer[1] << 8) | R_buffer[2]) & 0x3FF;